src/Avalonia.Controls/Window.cs

```diff
@@ -1185,6 +1228,11 @@ protected override void OnPropertyChanged(AvaloniaPropertyChangedEventArgs chang
                     impl.SetParent(_showingAsDialog ? newParent?.PlatformImpl! : (newParent?.PlatformImpl ?? null));
                 }
             }
+
+            else if (change.Property == CanResizeProperty)
+            {
+                CoerceValue(CanMaximizeProperty);
```

Comment:
Didn't know it's possible to trigger coercion of another property without changing it.